File: external/HDF-EOS/hdfeos/samples/InquireGrid.c
#include "hdf.h"
#include "HdfEosDef.h"

/*
 * In this example we will retrieve (1) information about the dimensions,
 * (2) the dimension mappings (geolocation relations), and (3) the grid
 * fields.
 */


main()
{

    intn            status, i;
    int32           gdfid, GDid1, ndim, nmap, nfld, rk, nt, nflds;
    int32           dims[32], rank[32], ntype[32];
    int32           n, strbufsize, sizes[16], GDid2;
    int32           xdimsize, ydimsize, dimsize, projcode, zonecode;
    int32           spherecode;
    
    float64         upleftpt[2], lowrightpt[2], projparm[16];
    
    char            dimname[1024], fieldlist[1024];


    /*
     * Open the Grid File for read only access
     */

    gdfid = GDopen("GridFile.hdf", DFACC_READ);

    
    if (gdfid != -1)
    {

	/* Attach the grid */

	GDid1 = GDattach(gdfid, "UTMGrid");
	GDid2 = GDattach(gdfid, "PolarGrid");
	
	ndim = GDinqdims(GDid1, dimname, dims);
	printf("Dimension list (UTMGrid): %s\n", dimname);
	for (i=0;i<ndim;i++) printf("dim size: %d\n", dims[i]);

	ndim = GDinqdims(GDid2, dimname, dims);
	printf("Dimension list (PolarGrid): %s\n", dimname);
	for (i=0;i<ndim;i++) printf("dim size: %d\n", dims[i]);


	dimsize = GDdiminfo(GDid1, "Time");
	printf("Size of \"Time\" Array: %d\n", dimsize);

	dimsize = GDdiminfo(GDid2, "Bands");
	printf("Size of \"Bands\" Array: %d\n", dimsize);


	
	status = GDgridinfo(GDid1, &xdimsize, &ydimsize,
			    upleftpt, lowrightpt);
	printf("X dim size, Y dim size (UTMGrid): %d %d\n", 
	       xdimsize, ydimsize);
	printf("Up left pt (UTMGrid): %lf %lf\n", 
	       upleftpt[0], upleftpt[1]);
	printf("Low right pt (UTMGrid): %lf %lf\n", 
	       lowrightpt[0], lowrightpt[1]);

	status = GDgridinfo(GDid2, &xdimsize, &ydimsize,
			    upleftpt, lowrightpt);
	printf("X dim size, Y dim size (PolarGrid): %d %d\n", 
	       xdimsize, ydimsize);
	printf("Up left pt (PolarGrid): %lf %lf\n", 
	       upleftpt[0], upleftpt[1]);
	printf("Low right pt (PolarGrid): %lf %lf\n", 
	       lowrightpt[0], lowrightpt[1]);
	
	       
	status = GDprojinfo(GDid1, &projcode, &zonecode,
			    &spherecode, NULL);
	printf("projcode , zonecode (UTMGrid): %d %d\n", projcode, zonecode);
	printf("spherecode (UTMGrid): %d\n", spherecode);	

	
	status = GDprojinfo(GDid2, &projcode, NULL,
			    &spherecode, projparm);
	printf("projcode (PolarGrid): %d\n", projcode);	
	printf("spherecode (PolarGrid): %d\n", spherecode);	
	for (i=0; i<13; i++)
	    printf("Projection Parameter: %d %lf\n",i,projparm[i]);
	
	nflds = GDinqfields(GDid1, fieldlist, rank, ntype);
	if (nflds != 0)
	{
	    printf("Data fields (UTMGrid): %s\n", fieldlist);
	    for (i=0;i<nflds;i++)
		printf("rank type: %d %d\n",rank[i],ntype[i]);
	}

	nflds = GDinqfields(GDid2, fieldlist, rank, ntype);
	if (nflds != 0)
	{
	    printf("Data fields (PolarGrid): %s\n", fieldlist);
	    for (i=0;i<nflds;i++)
		printf("rank type: %d %d\n",rank[i],ntype[i]);
	}
	
	
	status = GDfieldinfo(GDid2, "Spectra", rank, 
			     dims, ntype, dimname);
	printf("Spectra rank dims: %d\n",rank[0]);
	for (i=0; i<rank[0]; i++)
	    printf("Spectra dims: %d %d\n",i,dims[i]);
	printf("Spectra dims: %s\n", dimname);
	
	
	n = GDnentries(GDid1, HDFE_NENTDIM, &strbufsize);
	printf("Number of dimension entries (UTMGrid): %d\n", n);
	printf("Length of Dimension List (UTMGrid): %d\n", strbufsize);
	
	n = GDnentries(GDid1, HDFE_NENTDFLD, &strbufsize);
	printf("Number of data fields (UTMGrid): %d\n", n);
	printf("Length of Field List (UTMGrid): %d\n", strbufsize);
	
    }
    GDdetach(GDid1);
    GDdetach(GDid2);
    GDclose(gdfid);

    return;
}

File: external/HDF-EOS/hdfeos/samples/inquiregrid.f
	program inquiregrid

	integer      status, i, gddetach, gdclose, gdgridinfo
	integer      gdprojinfo
	integer*4    gdfid, gdid1, ndim, nflds, gddiminfo
	integer*4    dims(32), rank(32), ntype(32), gdopen, gdattach
	integer*4    n, strbufsize, gdid2, gdinqdims, gdinqflds
	integer*4    xdimsize, ydimsize, dimsize, projcode,zonecode
	integer*4    spherecode, origincode
    
	real*8       upleftpt(2), lowrightpt(2), projparm(13)
    
	character*72   dimname, fieldlist


	integer DFACC_READ
	parameter (DFACC_READ=1)


	
	gdfid = gdopen('GridFile.hdf', DFACC_READ)

    
	if (gdfid .ne. -1) then

	   gdid1 = gdattach(gdfid, 'UTMGrid')
	   gdid2 = gdattach(gdfid, 'PolarGrid')

	   ndim = gdinqdims(gdid1, dimname, dims)
	   write(*,*) 'Dimension list (UTMGrid): ', dimname
	   do i=1,ndim
	      write(*,*) 'dim size: ', dims(i)
	   enddo
	   
	   ndim = gdinqdims(gdid2, dimname, dims)
	   write(*,*) 'Dimension list (PolarGrid): ', dimname
	   do i=1,ndim
	      write(*,*) 'dim size: ', dims(i)
	   enddo
	   
	   dimsize = gddiminfo(gdid1, 'Time')
	   write(*,*) 'Size of "Time" Array: ', dimsize
	
	   dimsize = gddiminfo(gdid2, 'Bands')
	   write(*,*) 'Size of "Bands" Array: ', dimsize
	
	
	   status = gdgridinfo(gdid1, xdimsize, ydimsize,
     1                  	   upleftpt, lowrightpt)
	   write(*,*) 'X dim size, Y dim size (UTMGrid): ', 
     1 	               xdimsize, ydimsize
	   
	   write(*,*) 'Up left pt (UTMGrid): ', 
     1	               upleftpt(1), upleftpt(2)
	   write(*,*) 'Low right pt (UTMGrid): ', 
     1		       lowrightpt(1), lowrightpt(2)
	   
	   
	   status = gdgridinfo(gdid2, xdimsize, ydimsize,
     1                  	   upleftpt, lowrightpt)
	   write(*,*) 'X dim size, Y dim size (PolarGrid): ', 
     1 	               xdimsize, ydimsize
	   
	   write(*,*) 'Up left pt (PolarGrid): ', 
     1	               upleftpt(1), upleftpt(2)
	   write(*,*) 'Low right pt (PolarGrid): ', 
     1	               lowrightpt(1), lowrightpt(2)

	       
	   status = gdprojinfo(gdid1, projcode, zonecode,
     1	                       spherecode, projparm)
	   write(*,*) 'projcode , zonecode (UTMGrid): ', projcode,
     1	               zonecode
	   write(*,*) 'spherecode (UTMGrid): ', spherecode	

	
	   status = gdprojinfo(gdid2, projcode, zonecode
     1	                       spherecode, projparm)

	   status = gdprojinfo(gdid2, projcode, zonecode,
     1	                       spherecode, projparm)
	   
	   write(*,*) 'projcode (PolarGrid): ', projcode
	   write(*,*) 'spherecode (PolarGrid): ', spherecode
	   do i=1,13
	      write(*,*) 'Projection Parameter: ',i,projparm(i)
	   enddo
	   

	   nflds = gdinqflds(gdid1, fieldlist, rank, ntype)
	   if (nflds .ne. 0) then
	      write(*,*) 'Data fields (UTMGrid): ', fieldlist
	      do i=1,nflds
		 write(*,*) 'rank type: ',rank(i),ntype(i)
	      enddo
	   endif
	
	
	   nflds = gdinqflds(gdid2, fieldlist, rank, ntype)
	   if (nflds .ne. 0) then
	      write(*,*) 'Data fields (PolarGrid): ', fieldlist
	      do i=1,nflds
		 write(*,*) 'rank type: ',rank(i),ntype(i)
	      enddo
	   endif	
	
	
	   status = gdfldinfo(gdid2, 'Spectra', rank, 
     1	                        dims, ntype, dimname)
	   write(*,*) 'Spectra rank dims: ',rank(1)
	   do i=1,rank(1)
	      write(*,*) 'Spectra dims: ',i,dims(i), dimname
	   enddo
	
C	   n = gdnentries(gdid1, HDFE_NENTDIM, strbufsize)
C	   write(*,*) 'Number of dimension entries (UTMGrid): ', n
C	   write(*,*) 'Length of Dimension List (UTMGrid): ', strbufsize
	   
C	   n = gdnentries(gdid1, HDFE_NENTDFLD, strbufsize)
C	   write(*,*) 'Number of data fields (UTMGrid): ', n
C	   write(*,*) 'Length of Field List (UTMGrid): ', strbufsize
	   
	endif
	
	status = gddetach(gdid1)
	status = gddetach(gdid2)
	status = gdclose(gdfid)

	stop
	end
